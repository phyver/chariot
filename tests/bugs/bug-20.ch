codata pair('x,'y) where
    | Fst : pair('x,'y) -> 'x
    | Snd : pair('x,'y) -> 'y

data t('x) where
    | C : 'x -> t('x)

:set expand_clauses true
val h  { Fst = C y ; Snd = { Fst = C z ; Snd = zs }} = z

:set verbose 2
:show h
