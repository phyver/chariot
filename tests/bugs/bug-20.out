>> option verbose is set to 1
>> coinductive type pair was succesfully defined
>> inductive type t was succesfully defined
>> option expand_clauses is set to true
>> Typing for h successful
>> the definition for h is complete
>> the definition h is provably correct

>> option verbose is set to 2
val
   h : pair(t('a),pair(t('b),'c)) → 'b
    | h { Fst = C¹ x₂ ; Snd = { Fst = C¹ x₃ ; Snd = _₁ }² }⁰ = x₃
>> 
>> corresponding case and struct form:
>> x₁ → 
>>   case x₁.Fst of
>>     C x₂  ->  
>>       case x₁.Snd.Fst of
>>         C x₃  ->  x₃

